vmware-tanzu/tanzu-framework
# Decouple Feature-Flags from cli/runtime library and move to it to plugin specific code

### What this PR does / why we need it

- This PR decouples feature-flag definitions from the `cli/runtime` library and move it to plugin specific code
- This change uses `PluginDescriptor` for a specific plugin to set the defaults for the feature-flags and core CLI configures the default values during plugin installation or during plugin upgrade.
- The same default feature-flag values can also be configured with `tanzu config init` command if config file is updated manually and defaults needs to be set again.

### Which issue(s) this PR fixes


Fixes #3447

### Describe testing done for PR
- **Manual tests:**
  - Clean tanzu config file
  - Run `tanzu config get` command
    - It only adds tanzu cli specific default feature-flags to the config
  - Install all plugins with `tanzu plugin sync` to install all plugins
    - Run `tanzu config get` and verify plugin specific default feature-flags are configured correctly.
  - Manually update the config.yaml to remove certain default feature-flags for installed plugin and then run `tanzu config init`
    - Verify that all default feature-flags for installed plugin are configured as expected.

- **Unit tests** are written and validated as part of CI 



### Release note

```release-note
Decouple Feature-Flags from `cli/runtime` library and move to it to plugin specific code
```



### Additional information

#### Special notes for your reviewer
